src/inet.c

```
@@ -256,7 +256,14 @@ static int inet_pton6(const char *src, unsigned char *dst) {
       continue;
     }
     if (ch == '.' && ((tp + sizeof(struct in_addr)) <= endp)) {
-      int err = inet_pton4(curtok, tp);
+      int err;
+
+      /* Scope id is not allowed after ipv4 address */
+      pch = strchr(curtok, '%');
+      if (pch != NULL)
+        return UV_EINVAL;
+
+      err = inet_pton4(curtok, tp);
```

inet_pton4 will fail, I'd let it fail there.